crates/goose-mcp/src/developer/mod.rs

```diff
@@ -1011,6 +1011,15 @@ impl DeveloperRouter {
             let lines: Vec<&str> = content.lines().collect();
             let total_lines = lines.len();
 
+            if view_range.is_none() && total_lines > 2000 {
```

Comment:
Nit, I think documented `const LINE_LIMIT: usize = 2000` at the top would be nice